package/MDAnalysis/wham.py

```diff
@@ -0,0 +1,644 @@
+    numpad : int, optional
+        ['padding' values, for periodic PMFs; for nonperiodic use 0 (default)]
+    num_MC_trials : int, optional
+        Number of 'fake' data sets to create, if running bootstrap error analysis
+    metafile : str
+        [[ filename for writting metadata file ]]
+    outfile : str
+        [[ filename for writting final output ]]
+    check_only : bool
+        [[ (Default False). If True, only check values are valid; exit before 
+         running wham command ]]
+    run_boostrap : bool, optional
+        Whether to run Monte Carlo bootstrap error analysis.
+    calc_temperature, hist_min, hist_max
+        [[ TBA?, see wham above ]]
+    """
+    wham_args = locals()
+    ## TODO - need to provide path to runfile? where best to do this? (check
+    ## it's actually wham...)
+    wham_command = '~/wham/wham/wham'
+
+    # check everything here to catch issues before launching the command (and
+    # so we don't run something we don't want...)
+    _check_min_max(hist_min, 'hist_min', hist_max, 'hist_max')
+    _check_number(num_bins, 'number of bins', positive=True, integer=True)
+    _check_number(tol, 'tolerance', positive=True)
+    _check_number(calc_temperature, 'calculation temperature', positive=True)
+    _check_number(numpad, 'numpad', integer=True)
+    _check_number(num_MC_trials, 'num_MC_trials', positive=True, integer=True)
+    if periodicity not in ['', 'P', 'Ppi']: # OR 'P<val>'!
+        raise ValueError('Periodicity not valid...') ###
+        ## TODO - pass as something else? + Sort out P<val>
+    ## TODO - check files?
+
+    if not check_only:
+        # we should have a value for all args to run...
+        for key, item in wham_args.items():
+            if item is None:
+                raise TypeError("Must provide {}".format(key))
+        run_command = [wham_command, periodicity, hist_min, hist_max, num_bins, 
+                       tol, calc_temperature, numpad, metafile, outfile]
+        randSeed = 1 # TODO - how to deal with random seed - should make an argument?
+        if run_bootstrap:
+            run_command = run_command+[num_MC_trials, randSeed]
+        check_output(list_to_string(run_command), shell=True)
+        ## TODO - catch specific errors etc?
+
+
+
+def write_grossfield_input(bundle, filename='metadatafile.dat', 
+                           data_fname='timeseries_{}.dat', 
+                           data_names=None, ener_units=None, wham_units=None,
+                           checked=False, **kwargs):
+    """
+    [[ Write the appropriate metafile + timeseries files to run Grossfield wham
+    implementation on *bundle*.
+
+    Returns the min and max reaction coordinate values, set to the lowest/highest
+    value in the timeseries data if not set in args. ]]
+
+    [[ ARGS NEED SORTING ]]
+
+    Parameters
+    ----------
+    bundle
+    metafile_name
+    timeseriesfile_name
+    data_names
+    checked
+    **kwargs
```

Comment:
It would likely better to have a *write_kwargs*  parameter instead of `**kwargs`. This way, you know exactly what is done with your extra parameters, and you avoid letting typos go through. Typically, the way it is now, if I give a *data_name* parameter instead of /data_name**s**/ it may not raise an error.